src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,38 +2,44 @@
 
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
 
 public class SalaryInfo {
     private static final DateTimeFormatter DATE_FORMATTER
             = DateTimeFormatter.ofPattern("dd.MM.yyyy");
-    private static final int DATE_SUBSTRING_START = 0;
-    private static final int DATE_SUBSTRING_END = 10;
-    private static final int HOURS_INDEX = 2;
-    private static final int RATE_INDEX = 3;
+    private static final int DATA_COLUMN_INDEX = 0;
+    private static final int NAME_COLUMN_INDEX = 1;
+    private static final int HOURS_COLUMN_INDEX = 2;
+    private static final int RATE_COLUMN_INDEX = 3;
     private static final String COLUMNS_SEPARATOR = " ";
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        //I split the getSalaryInfo method into 4 separate methods
         int[] salary = salaryCounter(names, data, dateFrom, dateTo);
         return resultExecutor(names, salary, dateFrom, dateTo);
     }
 
     public int[] salaryCounter(String[] names, String[] data, String dateFrom, String dateTo) {
+        //Hi, in this solution I used the Arrays.asList() method,
+        //I know that this is a collection method, and we are not allowed to use collections
+        // in this task, but I don’t know how to solve this problem without using collections
+        //or less than three loops (and I think the method Arrays.asList().indexOf()
+        //actually uses a loop too). If you don't like this solution,
+        //you can look at the previous commit (called split method),
+        //it does not have collection methods,
+        //but it uses 3 loops (2 for counting and 1 for displaying information)
```

Comment:
the key to your solution is building a result string during the counting in the first 2 loops, because the third one is iterating over the names just like the first one, thus being redundant, also you should visit QnA for more details